src/config.rs

```diff
@@ -3,14 +3,41 @@
 
 use anyhow::{anyhow, Result};
 use std::collections::HashMap;
+use std::fs::File;
+use std::io::BufReader;
+
+pub const OCICRYPT_ENVVARNAME: &str = "OCICRYPT_KEYPROVIDER_CONFIG";
 
 /// DecryptConfig wraps the Parameters map that holds the decryption key
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
 pub struct DecryptConfig {
     /// map holding 'privkeys', 'x509s', 'gpg-privatekeys'
     pub param: HashMap<String, Vec<Vec<u8>>>,
 }
 
+/// Command describes the structure of command, it consist of path and args, where path defines the location of
+/// binary executable and args are passed on to the binary executable
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Command {
+    pub path: String,
+    pub args: Option<Vec<String>>,
+}
+
+/// KeyProviderAttrs describes the structure of key provider, it defines the different ways of invocation to key provider
+#[derive(Deserialize, Debug, Clone)]
+pub struct KeyProviderAttrs {
+    pub cmd: core::option::Option<Command>,
+    pub grpc: core::option::Option<String>,
+}
```

Comment:
one question i have is around the definition of the structures that get marshaled to/from JSON, like the `OcicryptConfig` object and related objects, is there a decorator - or somewhere we can specify like in golang the json key for consistency? it looks like `#[serde(alias = "key-providers")]` below does this? is it worthwhile to make this explicit for all structs?

Reply:
Yes serde alias macro as you mentioned above does that, I have made those changes now